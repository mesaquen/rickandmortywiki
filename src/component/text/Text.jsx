import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledText = styled.span`
  font-size: ${props => props.size};
  font-weight: ${props => (props.bold ? 'bold' : 'normal')};
  color: ${props => props.color};
`;

function Text(props) {
  return <StyledText {...props} />;
}

Text.propTypes = {
  size: PropTypes.string,
  color: PropTypes.string,
  bold: PropTypes.bool,
};

Text.defaultProps = {
  size: '16px',
  color: '#000000',
  bold: false,
};

export default memo(Text);
