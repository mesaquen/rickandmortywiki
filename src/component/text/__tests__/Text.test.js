import React from 'react';
import Renderer from 'react-test-renderer';
import Text from '../Text';
/**
 * @author Mesaque Francisco <mesaquenf@gmail.com>
 * @since 2019-11-16
 */

describe('Text', () => {
  let component = Renderer.create(<Text size="18px">Value</Text>);
  it('should have same snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
