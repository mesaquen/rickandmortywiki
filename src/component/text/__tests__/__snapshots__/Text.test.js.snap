// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should have same snapshot 1`] = `
<span
  className="sc-bdVaJa gUgWBB"
  color="#444"
  size="18px"
>
  Value
</span>
`;
